pulser/sequence.py

```diff
@@ -1081,29 +1081,49 @@ def abstract_repr(self) -> dict[str, Any]:
             ``serialize``
         """
 
-        res = {"register": [], "channels": [], "operations": []}
+        mapping = {
+            "CompositeWaveform": "composite",
+            "CustomWaveform": "custom",
+            "ConstantWaveform": "constant",
+            "RampWaveform": "ramp",
+            "BlackmanWaveform": "blackman",
+            "InterpolatedWaveform": "interpolated",
+            "KaiserWavewform": "kaiser",
+        }
+
+        res = {
+            "version": "1",
+            "register": {},
+            "channels": {},
+            "parameters": {},
+            "operations": [],
+        }
         operations = res["operations"]
         for call in self._calls:
             if call.name == "__init__":
                 register, device = call.args
                 # process register
                 res["device"] = device.name
                 for name, (x, y) in register.qubits.items():
-                    res["register"].append({"name": name, "x": x, "y": y})
+                    res["register"][name] = {"x": x, "y": y}
             elif call.name == "declare_channel":
                 ch_name, ch_kind = call.args
-                res["channels"].append({"name": ch_name, "kind": ch_kind})
+                res["channels"][ch_name] = {"hardware_channel": ch_kind}
+                if "initial_target" in call.kwargs:
+                    res["channels"][ch_name]["initial_target"] = call.kwargs[
+                        "initial_target"
+                    ]
```

Comment:
So we are storing the "initial_target" after all? Doesn't this make the following target operation redundant?